modules/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/NuxeoArtifactWebObject.java

```diff
@@ -96,4 +106,20 @@ public Object doViewDefault() {
     public String getSearchCriterion() {
         return String.format("'%s'", getNxArtifactId());
     }
+
+    protected String markdownToHTML(Blob readme) {
+        if (readme != null) {
+            try {
+                String content = readme.getString();
+                if (StringUtils.isNotBlank(content)) {
+                    return Framework.getService(HtmlSanitizerService.class)
```

Comment:
Do we really have to sanitize the output of the Markdown processor ? The processor itself doesn't do any sanitization ?

Reply:
I wasn't sure of the output, so i thought it was better to sanitize it. I can remove it if unnecessary.

Reply:
i confirm this is useful for xss attempts in the markdown text (and even if the library you suggest above can be configured to "sanitize URLs", they state in their README: "Note that this library doesn't try to sanitize the resulting HTML with regards to which tags are allowed, etc. That is the responsibility of the caller, and if you expose the resulting HTML, you probably want to run a sanitizer on it after this.")

Reply:
But the readmes that we convert too MarkDown are all originating from developer code, not from a user that would upload anything right? So that's not a threat model for which we would need to sanitize things.

Reply:
Ok, still we already sanitize a few things before rendering them in the context of explorer (typically content of "documentation" tags in XML code), i wanted to be consistent.
If considered unnecessary, again, i can remove it.

Reply:
Also, let's not forget that there's an import mechanism for all these snapshots, so in theory, a "tricked" user (who exported a distribution with XSS in its readmes, without realizing) could upload it on an explorer instance: maybe that matches a threat model that we'd like to avoid.

Reply:
Ok makes sense